src/sagas/index.js

```diff
@@ -183,6 +187,41 @@ export function* fetchByPage () {
   }))
 }
 
+function* fetchConsoles ({ payload: { vm } }) {
+  const vmId = vm.get('id')
+  const vmName = vm.get('name')
+  const apiConsoles = yield callExternalAction('consoles', Api.consoles, { payload: { vmId } })
+
+  // headless VM (no graphics_consoles defined) - API response is `{}`
+  if (Object.values(apiConsoles).length === 0) {
+    yield put(addUserMessage({ type: 'error', message: msg.consoleNotAvailableHeadless({ vmName }) }))
+    yield put(setVmConsoles({ vmId, consolesList: [], selectedConsole: EMPTY_CONSOLES_LIST }))
+  }
+
+  if (apiConsoles && apiConsoles.graphics_console) {
+    const defaultConsoleProtocol = yield select(state => state.config.get('defaultConsole'))
+    const consoles = Transforms.VmConsoles.toInternal({ consoles: apiConsoles })
+    // select the console to use based on availability and app/user configs
+    const selectedConsole =
+      consoles.length === 0 ? EMPTY_CONSOLES_LIST
+        : consoles.length === 1 ? consoles[0]
```

Comment:
@sjd78 I think that we should expect the supported only console to be used in case only one protocol is supported, regardless to default console. The default console is only for cases in which all options are supported and we need to auto choose for the user.

Reply:
@sgratch @sjd78 
I changed the code back to the prev version (now the system will use the only console available).